File: ui/composite/TrackArea.py
__author__ = 'toramisu'
from ui import *
from .Cursor import Cursor
from .Track import Track
from .TimestampBar import TimestampBar
from module.Events import *


class TrackArea(QWidget):
    def __init__(self, parent):
        super(TrackArea, self).__init__(parent)
        self.startFrameIdx = 1
        self.timestampBar = TimestampBar(self)
        self.timestampBar.resize(1280, 25)

        self.trackStack = QWidget(self)
        # self.trackStack.mousePressEvent = ignoreEvent
        self.vbox = QVBoxLayout(self.trackStack)
        self.vbox.setSpacing(0)
        self.trackStack.move(0, self.timestampBar.height())
        self.trackStack.resize(1280, 10)

        self.setMouseTracking(True)
        self.timestampBar.setMouseTracking(True)

        self.frameCursor = Cursor(self)
        self.frameCursor.attach(self.timestampBar)
        self.frameCursor.move(40, 0)
        self.tracks = []
        Event.add(SequencePlaybackEvent.RENDER_FRAME, self.onRenderFrame)

    def onRenderFrame(self, SequencePlaybackEvent):
        setX(self.frameCursor, SequencePlaybackEvent.frameIdx * 40)

    def resizeEvent(self, QResizeEvent):
        setHeight(self.frameCursor, self.height())

    def addTrack(self, trackInfo):
        track = Track()
        track.trackInfo = trackInfo
        track.load(trackInfo.frames)
        self.trackStack.resize(self.trackStack.width(), self.trackStack.height() + track.height())
        self.vbox.addWidget(track)
        self.tracks.append(track)
        pass

File: view/timeline/TrackArea.cpp
//
// Created by manoyuria on 2015/6/18.
//

#include "TrackArea.h"
#include "Track.h"

TrackArea::TrackArea(QWidget *parent) : QWidget(parent) {
    timestampBar = new TimestampBar(this);
    timestampBar->resize(1280, 25);


    trackStack = new QWidget(this);
    //todo make local
//    QVBoxLayout *vbox = new QVBoxLayout(trackStack);
    vbox = new QVBoxLayout(trackStack);
    vbox->setSpacing(0);
    trackStack->move(0, timestampBar->height());
    trackStack->resize(1280, 10);

    trackCursor = new TrackCursor(this);
    trackCursor->move(40, 0);
}

void TrackArea::add(TrackInfo *trackInfo) {
    Track *track = new Track();
    track->load(trackInfo);
    _setHeight(trackStack, trackStack->height() + track->height());
    qDebug()<<trackStack->height();
    vbox->addWidget(track);
}
